src/main/java/com/avispl/symphony/dal/communicator/aver/ptz/AverPTZCommunicator.java

```diff
@@ -0,0 +1,1885 @@
+	}
+
+	/**
+	 * This method is used to get the current display current gain level
+	 *
+	 * @return String This returns the gain level
+	 */
+	private String getGainLevel() {
+		try {
+			int currentSeqNum = ++sequenceNumber;
+			byte[] response = send(
+					buildSendPacket(cameraID, currentSeqNum, PayloadType.INQUIRY.getCode(), CommandType.INQUIRY.getCode(), PayloadCategory.CAMERA.getCode(), Command.GAIN_DIRECT.getCode()));
+
+			return String.valueOf(digestResponse(response, currentSeqNum, CommandType.INQUIRY, Command.GAIN_DIRECT));
+		} catch (Exception e) {
+			if (this.logger.isErrorEnabled()) {
+				this.logger.error("error during get gain level", e);
+			}
+		}
+		return LABEL_START_IRIS_VALUE;
+	}
+
+	/**
+	 * This method is used to get the current display current gain limit level
+	 *
+	 * @return String This returns the gain limit level
+	 */
+	private String getGainLimitLevel() {
+		try {
+			int currentSeqNum = ++sequenceNumber;
+			byte[] response = send(
+					buildSendPacket(cameraID, currentSeqNum, PayloadType.INQUIRY.getCode(), CommandType.INQUIRY.getCode(), PayloadCategory.CAMERA.getCode(), Command.GAIN_LIMIT_DIRECT.getCode()));
+			// Gain limit level: 24, 27, ... , 48. Value get from device: 0,1, ... , 8 => gain limit level = (value from device + 24 ) * 3
+			return String.valueOf(24 + (int) digestResponse(response, currentSeqNum, CommandType.INQUIRY, Command.GAIN_LIMIT_DIRECT) * 3);
+		} catch (Exception e) {
+			if (this.logger.isErrorEnabled()) {
+				this.logger.error("error during get gain limit level", e);
+			}
+		}
+		return LABEL_START_GAIN_LIMIT_VALUE;
+	}
+
+	/**
+	 * This method is used to get the current display current WB mode
+	 *
+	 * @return String This returns the WB mode
+	 */
+	private WBMode getWBMode() {
+		try {
+			int currentSeqNum = ++sequenceNumber;
+			byte[] response = send(
+					buildSendPacket(cameraID, currentSeqNum, PayloadType.INQUIRY.getCode(), CommandType.INQUIRY.getCode(), PayloadCategory.CAMERA.getCode(), Command.WB_MODE.getCode()));
+
+			WBMode mode = (WBMode) digestResponse(response, currentSeqNum, CommandType.INQUIRY, Command.WB_MODE);
+
+			if (mode == null) {
+				return WBMode.MANUAL;
+			} else {
+				return mode;
+			}
+		} catch (Exception e) {
+			if (this.logger.isErrorEnabled()) {
+				this.logger.error("error during get WB mode", e);
+			}
+			return WBMode.MANUAL;
+		}
+	}
+
+	/**
+	 * This method is used to get the current display current RGain value
+	 *
+	 * @return String This returns the RGain value
+	 */
+	private String getRGain() {
+		try {
+			int currentSeqNum = ++sequenceNumber;
+			byte[] response = send(
+					buildSendPacket(cameraID, currentSeqNum, PayloadType.INQUIRY.getCode(), CommandType.INQUIRY.getCode(), PayloadCategory.CAMERA.getCode(), Command.RGAIN_INQ.getCode()));
+
+			return String.valueOf(digestResponse(response, currentSeqNum, CommandType.INQUIRY, Command.RGAIN_INQ));
+
+		} catch (Exception e) {
+			if (this.logger.isErrorEnabled()) {
```

Comment:
Same here about having error enabled. No need for this check. Please check other places too :)